cumulusci/tasks/tests/test_package_2gp.py

```diff
@@ -0,0 +1,493 @@
+            json={"size": 1, "records": [{"Id": "0Ho000000000004AAA"}]},
+        )
+        responses.add(  # query for existing package version (unpackaged/pre)
+            "GET",
+            f"{self.base_url}/tooling/query/",
+            json={"size": 1, "records": [{"Id": "08c000000000004AAA"}]},
+        )
+        responses.add(  # check status of Package2VersionCreateRequest (unpackaged/pre)
+            "GET",
+            f"{self.base_url}/tooling/query/",
+            json={
+                "size": 1,
+                "records": [
+                    {
+                        "Id": "08c000000000004AAA",
+                        "Status": "Success",
+                        "Package2VersionId": "051000000000004AAA",
+                    }
+                ],
+            },
+        )
+        responses.add(  # get info from Package2Version (unpackaged/pre)
+            "GET",
+            f"{self.base_url}/tooling/query/",
+            json={
+                "size": 1,
+                "records": [
+                    {
+                        "SubscriberPackageVersionId": "04t000000000004AAA",
+                        "MajorVersion": 0,
+                        "MinorVersion": 1,
+                        "PatchVersion": 0,
+                        "BuildNumber": 1,
+                    }
+                ],
+            },
+        )
+        responses.add(  # create Package2VersionCreateRequest (main package)
+            "POST",
+            f"{self.base_url}/tooling/sobjects/Package2VersionCreateRequest/",
+            json={"id": "08c000000000002AAA"},
+        )
+        responses.add(  # check status of Package2VersionCreateRequest (main package)
+            "GET",
+            f"{self.base_url}/tooling/query/",
+            json={
+                "size": 1,
+                "records": [
+                    {
+                        "Id": "08c000000000002AAA",
+                        "Status": "Success",
+                        "Package2VersionId": "051000000000002AAA",
+                    }
+                ],
+            },
+        )
+        responses.add(  # get info from Package2Version (main package)
+            "GET",
+            f"{self.base_url}/tooling/query/",
+            json={
+                "size": 1,
+                "records": [
+                    {
+                        "SubscriberPackageVersionId": "04t000000000002AAA",
+                        "MajorVersion": 1,
+                        "MinorVersion": 0,
+                        "PatchVersion": 0,
+                        "BuildNumber": 1,
+                    }
+                ],
+            },
+        )
+        responses.add(  # get dependencies from SubscriberPackageVersion (main package)
+            "GET",
+            f"{self.base_url}/tooling/query/",
+            json={"size": 1, "records": [{"Dependencies": ""}]},
+        )
+
+        task.options["dependency_org"] = "2gp_dependencies"
+        task()
```

Comment:
Is there a meaningful assert to be added here?

Reply:
Fair question. I should at least make sure that all the fake responses were actually used.